File: frontend/lib/model/food.dart
class Food {
  final int foodId;
  final int userId;
  final int foodCategoryId;
  final String foodName;
  final String foodDescription;
  final double foodPrice;
  final bool isAvailable;
  final String createdAt;

  Food({
    required this.foodId,
    required this.userId,
    required this.foodCategoryId,
    required this.foodName,
    required this.foodDescription,
    required this.foodPrice,
    required this.isAvailable,
    required this.createdAt,
  });

  factory Food.fromJson(Map<String, dynamic> json) => Food(
        foodId: json['food_id'],
        userId: json['user_id'],
        foodCategoryId: json['food_category_id'],
        foodName: json['food_name'],
        foodDescription: json['food_description'],
        foodPrice: json['food_price'].toDouble(),
        isAvailable: json['is_available'],
        createdAt: json['created_at'],
      );

  Map<String, dynamic> toJson() => {
        'food_id': foodId,
        'user_id': userId,
        'food_category_id': foodCategoryId,
        'food_name': foodName,
        'food_description': foodDescription,
        'food_price': foodPrice,
        'is_available': isAvailable,
        'created_at': createdAt,
      };
}

File: backend/models/food.py
from sqlalchemy import Column, Integer, String, Float, Boolean, DateTime, ForeignKey
from sqlalchemy.sql import func
from core.db import Base

class Food(Base):
    __tablename__ = "foods"
    food_id = Column(Integer, primary_key=True, index=True)
    user_id = Column(Integer, nullable=False)
    food_category_id = Column(Integer, ForeignKey("categories.category_id"), nullable=False)
    food_name = Column(String, nullable=False)
    food_description = Column(String, nullable=False)
    food_price = Column(Float, nullable=False)
    is_available = Column(Boolean, default=True)
    created_at = Column(String, default=func.now())
